modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -23,6 +24,59 @@
 from pandas.core.dtypes.common import is_list_like
 
 
+def is_default_frame(value):
+    if isinstance(value, (tuple, list)):
+        result = False
+        for val in value:
+            result = result or is_default_frame(val)
+        return result
+    elif isinstance(value, dict):
+        return is_default_frame(list(value.values()))
+    else:
+        value = getattr(value, "_query_compiler", value)
```

Comment:
we're passing API level objects to backends when we're doing groupby on a list of Series whose parent is not grouping frame
```python
df = pd.DataFrame({"a": np.arange(3)})

s1, s2 = pd.Series(np.arange(3)), pd.Series(np.arange(3, 6))
df.groupby([s1, s2])
```

In that case at `df.groupby(...)` we will hit [this](https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/pandas/dataframe.py#L403-L420) if-else branch and will pass pure "by" with API-level Series to GroupByObject. 

GroupByObject constructor [do not](https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/pandas/groupby.py#L43-L92) convert high level objects to query compilers, and `_wrap_aggregation` method also [don't do](https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/pandas/groupby.py#L786-L795) that. 

So in groupby methods of query compiler, we can get API level objects as an arguments

Reply:
You're right. `_wrap_aggregation` uses API objects to default to pandas, but we should probably extract query compilers otherwise and avoid API objects in a backend (not in this PR though).